atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/impl/SnapshotTransaction.java

```diff
@@ -1058,11 +1058,10 @@ public void put(TableReference tableRef, Map<Cell, byte[]> values) {
     private void put(TableReference tableRef, Map<Cell, byte[]> values, long ttlDuration, TimeUnit ttlUnit) {
         Preconditions.checkArgument(!AtlasDbConstants.hiddenTables.contains(tableRef));
 
-        if (!validConflictDetection(tableRef)) {
-            conflictDetectionManager.recompute();
-            Preconditions.checkArgument(
-                    validConflictDetection(tableRef),
-                    "Not a valid table for this transaction.  Make sure this table name has a namespace: " + tableRef);
+        if (conflictDetectionManager.get(tableRef) == null) {
+            throw new IllegalArgumentException(
```

Comment:
Will this ever happen, given that you return RETRY_ON_WRITE_WRITE as a default value?